pegin/getaddress.go

```diff
@@ -0,0 +1,151 @@
+package pegin
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+
+	"github.com/btcsuite/btcutil"
+
+	"github.com/btcsuite/btcd/txscript"
+
+	"github.com/vulpemventures/go-elements/network"
+	"github.com/vulpemventures/go-elements/payment"
+
+	"github.com/btcsuite/btcd/chaincfg"
+
+	"github.com/vulpemventures/go-elements/address"
+
+	"github.com/btcsuite/btcd/btcec"
+)
+
+const (
+	MainNet NetworkType = iota
+	RegtestNet
+)
+
+type NetworkType int
+
+type AddressInfo struct {
+	ClaimScript      string
+	MainChainAddress string
+}
+
+// GetAddressInfo simulates getpeginaddress RPC call which returns mainChain
+// address to which BTC should be sent in BTC network and claim script that is
+// to be used in claimpegin RPC alongside with bitcoinTx and txoutproof
+func GetAddressInfo(
+	publicKey []byte,
+	fedpegInfo FedpegInfo,
```

Comment:
actually, we only need the `FedPegInfo.FedPegScript` so the federation's script ([]byte) could replace `FedpegInfo` parameter

Reply:
I would remove GetAddressInfo entirerely actually, ClaimWitnessScript and PeginAddress are enough high level API